File: Selenium/Activities/Activity9.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.io.IOException;
import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) throws IOException {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/ajax");
        System.out.println("Title of webpage:" + driver.getTitle());

        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.findElement(By.xpath("//button[@class='ui violet button']")).click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//h1[text()='HELLO!']")));
        System.out.println(driver.findElement(By.tagName("h3")).getText());
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"),"I'm late!"));
        System.out.println(driver.findElement(By.tagName("h3")).getText());
        driver.close();
    }
}

File: Selenium/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/ajax")
    wait = WebDriverWait(driver, 10)
    driver.find_element(By.XPATH, "//button[@class='ui violet button']").click()
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH, "//h1[text()='HELLO!']")))
    print(driver.find_element(By.TAG_NAME, "h3").text)
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h3"), "I'm late!"))
    print(driver.find_element(By.TAG_NAME, "h3").text)
    driver.close()
